src/corehost/cli/fxr/fx_ver.cpp

```diff
@@ -108,20 +115,125 @@ int fx_ver_t::compare(const fx_ver_t&a, const fx_ver_t& b)
         return (a.m_patch > b.m_patch) ? 1 : -1;
     }
 
-    if (a.m_pre.empty() != b.m_pre.empty())
+    if (a.m_pre.empty() || b.m_pre.empty())
     {
-        // Either a is empty or b is empty
-        return a.m_pre.empty() ? 1 : -1;
+        // Either a is empty or b is empty or both are empty
+        return a.m_pre.empty() ? !b.m_pre.empty() : -1;
     }
 
-    // Either both are empty or both are non-empty (may be equal)
-    int pre_cmp = a.m_pre.compare(b.m_pre);
-    if (pre_cmp != 0)
+    // Both are non-empty (may be equal)
+    size_t idStart = 1;
```

Comment:
Please add a comment (or maybe even assert) that the first chracter in pre is the hyphen.